flink-core/src/main/java/org/apache/flink/api/connector/sink2/Sink.java

```diff
@@ -55,17 +53,11 @@ public interface Sink<InputT> extends Serializable {
      * @return A sink writer.
      * @throws IOException for any failure during creation.
      */
-    SinkWriter<InputT> createWriter(InitContext context) throws IOException;
+    SinkWriter<InputT> createWriter(WriterInitContext context) throws IOException;
 
     /** The interface exposes some runtime info for creating a {@link SinkWriter}. */
     @PublicEvolving
-    interface InitContext {
-        /**
-         * The first checkpoint id when an application is started and not recovered from a
-         * previously taken checkpoint or savepoint.
-         */
-        long INITIAL_CHECKPOINT_ID = 1;
-
+    interface WriterInitContext extends InitContext {
```

Comment:
I'm afraid this change may cause some compatibility issues for connectors.
WDYT? @dannycranmer, @MartijnVisser

Reply:
That's why we had a FLIP and a vote @Jiabao-Sun

Reply:
Maybe I misunderstood, but I didn't see any proposals regarding changes to Sink and InitContext in the FLIP-371 document. 
Would introducing CommitterInitContext inheriting InitContext meet the requirements?

Reply:
We discussed the possibility of `CommitterInitContext` inheriting from the `InitContext`, but that would have been a short term solution only and the resulting API would be inconsistent. That is why the decision was to have `WriterInitContext` and `CommitterInitContext`, and accept that the connectors has to update the code.

If you are interested in the Source related changes please follow FLIP-372 too, as there will be similar changes there.